src/main/scala/org/constellation/storage/NotificationService.scala

```diff
@@ -1,14 +1,19 @@
 package org.constellation.storage
 import cats.effect.IO
 import org.constellation.primitives.PeerNotification
+import org.constellation.storage.algebra.{Lookup, MerkleStorageAlgebra}
 
-class NotificationService(size: Int = 2000)  extends MerkleService[String, PeerNotification]{
-  val merklePool = new StorageService[Seq[String]]()
-  val memPool = new StorageService[PeerNotification]()
+class NotificationService(size: Int = 2000)  extends MerkleStorageAlgebra[IO, String, PeerNotification]{
```

Comment:
I realize you just refactored from the old one -- but we can get rid of size here I believe